batchflow/models/tf/encoder_decoder.py

```diff
@@ -233,14 +258,20 @@ def encoder(cls, inputs, name='encoder', **kwargs):
 
                 for i in range(steps):
                     with tf.variable_scope('encoder-'+str(i)):
-                        # Preprocess tensor with given block
-                        args = {**kwargs, **block_args, **unpack_args(block_args, i, steps)} # enforce priority of keys
-                        x = base_block(x, name='pre', **args)
-
-                        # Downsampling
-                        if downsample.get('layout') is not None:
-                            args = {**kwargs, **downsample, **unpack_args(downsample, i, steps)}
-                            x = conv_block(x, name='downsample-{}'.format(i), **args)
+                        # Make all the args
+                        args = {**kwargs, **block_args, **unpack_args(block_args, i, steps)}
+                        downsample_args = {**kwargs, **downsample, **unpack_args(downsample, i, steps)}
+
+                        if order in ['bd', 'bp']: # block -> downsample
```

Comment:
what is `'p'`?
what about checking `order[0] == 'b'`?

Reply:
`p` stands for `pooling`